### PR TITLE
Fix for multiple ip addresses on an interface which are in the same subnet.

### PR DESCRIPTION
**Pull request by BaronGreenback**
**Bug:** If an interface has multiple IP addresses in the same subnet, the system selected the first it encountered.

This fix permits multiple ip addresses in the same subnet to be present on an interface.
eg. 192.168.1.10/24, 192.168.1.20/24

This has been added as a check in the test unit.

### CODE REVIEW

**Review by crobibero — APPROVED** [review]

File: Jellyfin.Networking/Manager/NetworkManager.cs [hunk 1]
```diff
@@ -157,15 +157,16 @@ public NetworkManager(IConfigurationManager configurationManager, ILogger<Networ
         /// Creates a new network collection.
         /// </summary>
         /// <param name="source">Items to assign the collection, or null.</param>
+        /// <param name="unique"><c>True</c> if subnets that overlap should be merged (default).</param>
         /// <returns>The collection created.</returns>
-        public static Collection<IPObject> CreateCollection(IEnumerable<IPObject>? source = null)
+        public static Collection<IPObject> CreateCollection(IEnumerable<IPObject>? source = null, bool unique = true)
```
Comment (cvium) [inline Jellyfin.Networking/Manager/NetworkManager.cs:162]: There's 5 usages of `CreateCollection` in NetworkManager and 1 in DlnaEntryPoint. All 5 in NetworkManager are called with false. Is it the correct default then?

Comment (BaronGreenback) [reply Jellyfin.Networking/Manager/NetworkManager.cs]: Thought there were more.

Have removed the extra param from CreateCollection, and hardcoded value in addItem.

File: MediaBrowser.Common/Net/NetworkExtensions.cs [hunk 2]
```diff
@@ -27,9 +27,10 @@ public static void AddItem(this Collection<IPObject> source, IPAddress ip)
         /// </summary>
         /// <param name="source">The <see cref="Collection{IPObject}"/>.</param>
         /// <param name="item">Item to add.</param>
-        public static void AddItem(this Collection<IPObject> source, IPObject item)
+        /// <param name="unique"><c>True</c> if subnets that overlap should be merged (default).</param>
+        public static void AddItem(this Collection<IPObject> source, IPObject item, bool unique = true)
         {
-            if (!source.ContainsAddress(item))
+            if (!source.ContainsAddress(item) || !unique)
```
Comment (cvium) [inline MediaBrowser.Common/Net/NetworkExtensions.cs:33]: Does this check that the IP is not the exact same?

Comment (BaronGreenback) [reply MediaBrowser.Common/Net/NetworkExtensions.cs]: changed the var name and description.

True means 10.1.1.1/24 and 10.1.1.2/24 would both be added as they are addresses.
False means 10.1.1.1/24 and 10.1.1.2/24 would not add the 2nd as the first will be treated as the network 10.1.1.0/24 of which 10.1.1.2/24 is part.

**Review by cvium — APPROVED** [review]

**Pull request merged**